gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxForm,
+  NxFormGroup,
+  NxToggle,
+  useToggle,
+  NxFieldset,
+  NxRadio,
+  NxP,
+  NxCode
+} from '@sonatype/react-shared-components';
+import classnames from 'classnames';
+
+const DarkModeModal = () => {
+  const [showModal, setShowModal] = useState(false);
+  const modalCloseHandler = () => setShowModal(false);
+
+  const [enableModeChange, toggleEnableModeChange] = useToggle(false);
+  const [mode, setMode] = useState<string | null>(null);
+  const [radioDisabled, setRadioDisabled] = useState(true);
+
+  const htmlElement = document.documentElement;
+  const htmlClasses = classnames('nx-html', 'nx-html--page-scrolling',
+      {'nx-html--enable-color-schemes': enableModeChange,
+        'nx-html--light-mode': mode === 'light' && enableModeChange,
+        'nx-html--dark-mode': mode === 'dark' && enableModeChange
+      });
+
+  useEffect(()=> {
+    htmlElement.className = htmlClasses;
+  }, [enableModeChange, mode]);
+
+  useEffect(()=> {
+    enableModeChange ? setRadioDisabled(false) : setRadioDisabled(true);
+  }, [enableModeChange]);
+
+  // const { classList } = document.documentElement;
+
+  // useEffect(()=>{
+  //   if (mode === 'light') {
+  //     classList.remove('nx-html--dark-mode');
+  //     classList.add('nx-html--light-mode');
+  //   }
+  //   else if (mode === 'dark') {
+  //     classList.remove('nx-html--light-mode');
+  //     classList.add('nx-html--dark-mode');
+  //   }
+  //   else {
+  //     classList.remove('nx-html--dark-mode', 'nx-html--light-mode');
+  //   }
+  // }, [enableModeChange, mode]);
+
+  // useEffect(()=> {
+  //   if (enableModeChange) {
+  //     classList.add('nx-html--enable-color-schemes');
+  //     setRadioDisabled(false);
+  //   }
+  //   else if (!enableModeChange) {
+  //     classList.remove('nx-html--enable-color-schemes', 'nx-html--dark-mode', 'nx-html--light-mode');
+  //     setRadioDisabled(true);
+  //   }
+  // }, [enableModeChange]);
+
+  return (
+    <>
+      <NxButton variant="tertiary" onClick={() => setShowModal(true)}>Light / Dark Mode</NxButton>
+      { showModal &&
+        <NxModal id="nx-modal-dark-mode-example"
+                 role="alertdialog"
+                 onCancel={modalCloseHandler}
+                 aria-label="NxModal to set light or dark mode">
```

Comment:
Use `aria-labelledby` to reference the `h2` instead of `aria-label`

Reply:
> Use `aria-labelledby` to reference the `h2` instead of `aria-label`

changed to `aria-lablledby` here: [92a7597](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/92a75978913d80a5b8101f5aef71e4c7307c81e7)

Reply:
> changed to aria-lablledby here: 92a7597

The id in question should really be more specific to this modal too

Reply:
> The id in question should really be more specific to this modal too

Renamed here: [227bab8](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/227bab8fc6853c6e335f2877b23e1b86b482bee5)